WordPress/Classes/Services/PostService.m

```diff
@@ -234,6 +254,20 @@ - (void)syncPostsOfType:(PostServiceType)postType
 - (void)uploadPost:(AbstractPost *)post
            success:(void (^)(AbstractPost *post))success
            failure:(void (^)(NSError *error))failure
+{
+    [self uploadPost:post
+      creationOption:CreationOptionAsIs
+             success:success
+             failure:failure];
+}
+
+/**
+ * @param creationOption The strategy to use if [remote createPost:] is invoked.
+ */
+- (void)uploadPost:(AbstractPost *)post
+    creationOption:(CreationOption)creationOption
```

Comment:
I started with `forceDraft` but thought it would be confusing because `uploadPost` does both **updating** and **creating**. A `true` `forceDraft` could mean that even if we are just updating the post, we would change the post to a draft. Furthermore, a `false` `forceDraft` is also not clear in what it does without reading the code.  

This is why I opted to use an enum instead to clarify what to use _if creating_ a post: (a) change it to a draft, or (b) leave as is. I believe this, contrary to your argument of “making our code more explicit“, is very explicit. 🙂 

But I'd like to move on and I slightly agree that the enum is still confusing and even over-engineered. It's sort of a compromise instead of refactoring a whole lot of this class. So what do you think about `forceDraftIfCreating` instead? I think I'll have to add more comments to describe the effects of the boolean but it is what is. 😬

Reply:
I like `forceDraftIfCreating`. :)

Reply:
Done in c3474abb01b9ba92e1009d2a23a9f72fe94e3ec3.